docs/content/docs/data-structures/sequence.md

```diff
@@ -0,0 +1,241 @@
+    let props5 = sharedString.getPropertiesAtPosition(5);
+    // props1 = {}
+    // props5 = {}
+
+    // set property called weight on positions 0 and 1
+    sharedString.annotateRange(0, 2, { weight: 5 });
+    props1 = sharedString.getPropertiesAtPosition(1);
+    props5 = sharedString.getPropertiesAtPosition(5);
+    // props1 = { weight: 5 }
+    // props5 = {}
+
+    // set property called decoration on all positions
+    sharedString.annotateRange(
+        0,
+        sharedString.getLength(),
+        { decoration: "underline" });
+    props1 = sharedString.getPropertiesAtPosition(1);
+    props5 = sharedString.getPropertiesAtPosition(5);
+    // props1 = { weight: 5, decoration: "underline" }
+    // props5 = { decoration: "underline" }
+
+    // remove property called weight on all positions
+    sharedString.annotateRange(
+        0,
+        sharedString.getLength(),
+        { weight: null });
+    props1 = sharedString.getPropertiesAtPosition(1);
+    props5 = sharedString.getPropertiesAtPosition(5);
+    // props1 = { decoration: "underline" }
+    // props5 = { decoration: "underline" }
+```
+
+Whenever an operation is performed on a sequence a *sequenceDelta* event will be raised. This even provides the ranges
+affected by the operation, the type of the operation, and the properties that were changes by the operation.
+
+## How Collaboration Works
+
+Like other data structures the sequences are eventually consistent which means all collaborators will end up in the same
+final state, however, the intermediate states seen by each collaborator may not be seen by other collaborators. These
+intermediate states occur when two or more collaborators modify the same position in the sequence which results in a
+conflict.
+
+The basic strategy for insert conflict resolution in the sequence is to merge *far*. This strategy depends on a
+fundamental property of the Fluid Framework, which is guaranteed ordering. So, if two or more collaborators perform an
+operation on a sequence, the operations will be given an ordering and all clients will see those operations in the same
+order. What this means for the merge *far* strategy for resolving conflicting inserts is that the first operation will
+be placed in the conflicting position when it is received. When the next insert with the same position arrives and is
+applied it will be placed at the specified position and the previous inserts content position will be increased by the
+length of the incoming content pushing is farther towards the length of the sequence. This is what we call merging
+*far*.
+
+Like insert the strategies for remove and annotate also rely on guaranteed ordering. For remove and annotate only
+content visible to the collaborator creating the operation will be modified, any content ordered after the won't be.
+
+For remove this means we can't have an insert and a remove at the same time, as they will have an order, and all
```

Comment:
> For remove this means we can't have an insert and a remove at the same time, as they will have an order, and all collaborators will see the operations in the same order

I'm not sure what exactly this sentence tries to say :)

Reply:
This is trying to say that the concept of "at the same time" doesn't exist in Fluid since ops are explicitly ordered.

Reply:
I do not think Fluid brings anything unique here.
Most processors (Intel x86 family including) have global ordering guarantee, i.e. all parties (threads) observe events in same order. I.e. if one party (thread) saw A happen after B, then all other parties/threads will see these events in exactly same order.
But in multi-threading, we use "at the same time" terminology (or rather more correct - "concurrent" changes), despite the fact that there will be order assigned to those events.

If we want to not use "at the same time" terminology, then we need to come up with some other terminology that describes changes that happened so that each of it had no chance to observe the other at the time change was made (aka concurrent changes).